admin/src/js/services/select2-search.js

```diff
@@ -186,16 +246,30 @@ angular.module('inboxServices').factory('Select2Search',
 
             if (docId) {
               getDoc(docId).then(function(doc) {
-                selectEl.select2('data')[0].doc = doc;
-                selectEl.trigger('change');
+                const select2Data = selectEl.select2('data') || [];
+                const selected = select2Data.find((d) => d.id === docId);
+                if (selected) {
+                  selected.doc = doc;
+                } else {
+                  select2Data.push({
+                    id: docId,
+                    doc: doc,
+                    text: doc.name,
+                  });
+                }
+                selectEl.select2('data', select2Data);
+                $timeout(() => { //NoSONAR
+                  selectEl.trigger('change');
+                }, 1000);
+
+                return selectEl;
```

Comment:
It looks like this block is very similar to a block above (line 180). I think we should find a way to reuse this code.